### PR TITLE
[ feature ]: Added priority search doc. in searching-algorithm folder

### LINKED ISSUE #1951: [Feature]: Want to add priority search algorithm in algorithm folder
### Feature Name

priority search algorithm

### Feature Description

Priority search is an algorithmic approach commonly used in scenarios where elements have different priorities and need to be accessed or processed based on those priorities. The goal is to retrieve the highest (or lowest) priority item efficiently, which is particularly useful in scheduling, pathfinding, and resource allocation.

### Motivation

A priority search algorithm is designed to quickly retrieve or process high-priority items, making it ideal for tasks like scheduling and pathfinding. It helps users save time, boosts responsiveness, and improves the user experience by focusing on the most relevant or urgent information first.

### Implementation Suggestions (Optional)

- **Choose a Data Structure:** Use a priority queue (often a Min/Max Heap) to manage elements based on priority.
- **Define Priority Rules:** Establish how priorities are assigned to elements (e.g., smallest number, shortest path).
- **Insert Elements:** Add items to the priority queue with their associated priorities.
- **Retrieve Top Priority:** Use the queue to get the highest-priority item efficiently.
- **Repeat or Terminate:** Continue retrieval until all desired elements are processed or the condition is met.

### Feature Type

New Algorithm

### Does this feature require additional resources?

- [ ] Additional Documentation
- [ ] Third-Party Libraries
- [ ] UI/UX Design
- [X] Other

### References (Optional)

https://en.wikipedia.org/wiki/Priority_search_tree

### PR DESCRIPTION
**Pull request by Jivan052**
### Description:
Added Priority search algorithm doc. in searching-algorithm folder which is part of algorithms folder.

**Description of algo:**
Priority Search efficiently finds the closest rectangle to a query point in a 2D plane by sorting rectangles based on priority. It offers fast search times when the data is sorted but can degrade to linear time in the worst case. It's commonly used in GIS, graphics, and robotics for spatial queries.

Dependencies: NA
Fixes: #1951

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/algorithms/Searching Algorithms/PrioritySearch.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
+---
+
+id: Priority-search-algorithm
```
Comment (ajay-dhangar) [inline docs/algorithms/Searching Algorithms/PrioritySearch.md]: replace `id: Priority-search-algorithm` with `id: priority-search-algorithm`

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

File: docs/algorithms/Searching Algorithms/PrioritySearch.md (hunk 1 above)
Comment (Jivan052) [reply docs/algorithms/Searching Algorithms/PrioritySearch.md]: changes has been made. review it.

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**